code/modules/mob/living/carbon/xenomorph/castes/pyrogen/abilities_pyrogen.dm

```diff
@@ -0,0 +1,529 @@
+			fired.apply_status_effect(STATUS_EFFECT_MELTING_FIRE, PYROGEN_FIREBALL_MELTING_STACKS)
+			fired.take_overall_damage(fire_damage, BURN, FIRE, FALSE, FALSE, TRUE, 0)
+	new /obj/effect/temp_visual/xeno_fireball_explosion(get_turf(target))
+
+/datum/ammo/xeno/fireball/on_hit_turf(turf/target, obj/projectile/proj)
+	. = ..()
+	var/fire_damage = 0
+	var/atom/target_atom = target.density ? proj : target
+	for(var/turf/affecting AS in RANGE_TURFS(1, target_atom))
+		new /obj/melting_fire(affecting)
+		for(var/mob/living/carbon/fired in affecting)
+			fire_damage = PYROGEN_FIREBALL_AOE_DAMAGE
+			if(fired.has_status_effect(STATUS_EFFECT_MELTING_FIRE))
+				var/datum/status_effect/stacking/melting_fire/debuff = fired.has_status_effect(STATUS_EFFECT_MELTING_FIRE)
+				debuff.add_stacks(PYROGEN_FIREBALL_MELTING_STACKS)
+			fired.apply_status_effect(STATUS_EFFECT_MELTING_FIRE, PYROGEN_FIREBALL_MELTING_STACKS)
+			fired.take_overall_damage(fire_damage, BURN, FIRE, FALSE, FALSE, TRUE, 0)
+	new /obj/effect/temp_visual/xeno_fireball_explosion(get_turf(target_atom))
+
+/// Needed cause AMMO_HIT_TURF does not call ON_HIT_TURF for some reason. calls this.
+/datum/ammo/xeno/fireball/do_at_max_range(turf/target, obj/projectile/proj)
+	. = ..()
+	var/fire_damage = 0
+	var/atom/target_atom = target.density ? proj : target
+	for(var/turf/affecting AS in RANGE_TURFS(1, target_atom))
+		new /obj/melting_fire(affecting)
+		for(var/mob/living/carbon/fired in affecting)
+			fire_damage = PYROGEN_FIREBALL_AOE_DAMAGE
+			if(fired.has_status_effect(STATUS_EFFECT_MELTING_FIRE))
+				var/datum/status_effect/stacking/melting_fire/debuff = fired.has_status_effect(STATUS_EFFECT_MELTING_FIRE)
+				debuff.add_stacks(PYROGEN_FIREBALL_MELTING_STACKS)
+			fired.apply_status_effect(STATUS_EFFECT_MELTING_FIRE, PYROGEN_FIREBALL_MELTING_STACKS)
+			fired.take_overall_damage(fire_damage, BURN, FIRE, FALSE, FALSE, TRUE, 0)
+	new /obj/effect/temp_visual/xeno_fireball_explosion(get_turf(proj))
+
+
+// ***************************************
+// *********** Firenado
+// ***************************************
+/datum/action/ability/activable/xeno/firestorm
+	name = "Fire Storm"
+	action_icon_state = "whirlwind"
+	desc = "Unleash 3 fiery tornados. They will try to close on your target tile"
+	target_flags = ABILITY_TURF_TARGET
+	ability_cost = 50
+	cooldown_duration = 8 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_FIRENADO,
+	)
+
+/obj/effect/xenomorph/firenado
+	name = "Plasma Whirlwind"
+	desc = "A glowing whirlwind of... cold plasma? seems to \"burn\" "
+	icon = 'icons/effects/64x64.dmi'
+	icon_state = "whirlwind"
+	anchored = TRUE
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	pixel_x = -16
+	/// How many process ticks can we do before we are deleted/we extinguish
+	var/burn_ticks_left = 20
+	/// Target turf to bias going towards
+	var/turf/target
+
+/obj/effect/xenomorph/firenado/New(loc, arg_target)
+	. = ..()
+	target = arg_target
```

Comment:
use inititalize

Reply:
initialize doesn't pass arguments from New() does it ?

Reply:
yes it does